pinot-core/src/test/java/org/apache/pinot/core/query/optimizer/filter/TimePredicateFilterOptimizerTest.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.optimizer.filter;
+
+import java.util.List;
+import org.apache.pinot.common.request.Expression;
+import org.apache.pinot.common.request.Function;
+import org.apache.pinot.core.query.optimizer.filter.utils.Range;
+import org.apache.pinot.pql.parsers.pql2.ast.FilterKind;
+import org.apache.pinot.sql.parsers.CalciteSqlParser;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+
+public class TimePredicateFilterOptimizerTest {
+  private static final TimePredicateFilterOptimizer OPTIMIZER = new TimePredicateFilterOptimizer();
+
+  @Test
+  public void testTimeConvert() {
+    testTimeConvert("timeConvert(col, 'MILLISECONDS', 'MILLISECONDS') > 1620830760000",
+        new Range(1620830760001L, true, null, false));
```

Comment:
I think this should made to be `new Range(1620830760000L, false, null, false))`

Reply:
These 2 ranges are equivalent

Reply:
Yes, but my point is that we shouldn't be changing input values unless absolutely needed. In this case the +1 can be easily avoided.

Reply:
Also it seems like, `timconvert(col, 'MILLISECONDS', 'MILLISECONDS') > 9223372036854775807` (long max value) will turn into `col >= 9223372036854775807` which seems incorrect. A test case on this would be useful.

Reply:
The algorithm is to handle the general case, so as long as the optimized one is correct we should be good.
For the super large/small value, I'll add a check that the millis is in valid range. If not, the optimizer won't change the filter.

Reply:
I think it would be much simpler and accurate if you don't do +1. That way it will handle all cases and you won't need to add a if statement for large values.

Reply:
We can avoid the `+1` only when the output format is `MILLISECONDS`, and it will complicate the step 2 (convert millis range to input range). Since they are equivalent, I'd prefer just keeping it simple. Added a check that millis must be in valid range to avoid unexpected overflow/underflow.

Reply:
I see, 2071-1-1 is the current max timestamp. I guess I can live with that. Although it seems like all the semantic checks should be done outside and before the optimizers are run (but that could be refactored at a later time).